util/reggen/countermeasure.py

```diff
@@ -0,0 +1,80 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, List
+
+from .lib import check_keys, check_str, check_list
+
+# TO BE DEFINED
+CM_ASSETS = [
+    'FOO'
+]
+
+# TO BE DEFINED
+CM_TYPES = [
+    'BAR'
+]
+
+
+class CounterMeasure:
+    """Object holding details for one countermeasure within an IP block."""
+
+    def __init__(self, asset: str, cm_type: str, desc: str):
+        self.asset = asset
+        self.cm_type = cm_type
+        self.desc = desc
+
+    @staticmethod
+    def from_raw(what: str, raw: object) -> 'CounterMeasure':
+        """
+        Create a CounterMeasure object from a dict.
+
+        The dict must have the keys 'name' and 'desc', where 'name' has
+        to follow the canonical countermeasure naming convention.
+        """
+        rd = check_keys(raw, what, ['name', 'desc'], [])
+
+        name = check_str(rd['name'], f'name field of {what}')
+        desc = check_str(rd['desc'], f'desc field of {what}')
+
+        try:
+            asset, cm_type = name.split('.')
+        except ValueError:
+            raise ValueError(
+                f'Invalid countermeasure format: {name} ({what}).')
+
+        if asset not in CM_ASSETS:
+            raise ValueError(
+                f'Invalid countermeasure format: {name} ({what}).')
```

Comment:
Perhaps clearer to say `Invalid asset` instead, since this checks the validity of the asset specifically. Likewise below.

Reply:
I agree, also because it means that the different error messages are distinct, which makes debugging things a bit easier.

Reply:
Ah yeah agreed. I mistakenly changed this when shortening the error messages. Done!